ablator/main/state.py

```diff
@@ -290,6 +548,19 @@ def _init_trials(self, resume: bool = False) -> list[ParallelConfig]:
         return trials
 
     def sample_trials(self, n_trials_to_sample: int) -> list[ParallelConfig] | None:
+        """
+        Sample n trials from the search space and update database.
```

Comment:
Should we mention that this depends on the number of trials remaining? e.g number of sampled trials = number of trials remaining if `n_trials_to_sample` exceeds `n_trials_remaining`, or zero if there's out of trials remaining